exercises.py

```diff
@@ -1,13 +1,394 @@
+
+# 13. Write a short program that prints the numbers 1 to 10 using a for loop. Then write an equivalent program that prints the numbers 1 to 10 using a while loop.
+for i in range (1,11):
+    print(i)
+
+number = 0
+while number < 10 :
+    number = number + 1
+    print(number)
+
+# 14. If you had a function named bacon() inside a module named spam, how would you call it after importing spam?
+import spam
+
+spam.bacon()
+
+# Extra credit: Look up the round() and abs() functions on the internet, and find out what they do. Experiment with them in the interactive shell.
+###################################################################################################################
+
+
+
+
+
+
 
-# Chapter 1
 
-# Chapter 2
 
 # Chapter 3
 
+
+# 1. Why are functions advantageous to have in your programs?
+to reduce repetition, and make your programs shorter and easier to read
+
+# 2. When does the code in a function execute: when the function is defined or when the function is called?
+when the function is called
+
+# 3. What statement creates a function?
+def
+
+# 4. What is the difference between a function and a function call?
+a function on its own just exists and does nothing until called
+calling a function runs it
+
+# 5. How many global scopes are there in a Python program? How many local scopes?
+one global, and perhaps infinite local scopes
+
+# 6. What happens to variables in a local scope when the function call returns?
+it is destroyed
+
+# 7. What is a return value? Can a return value be part of an expression?
+the value that a function call evaluates.  yes, because expressions are composed of values and operators. and a return value is a value
+
+# 8. If a function does not have a return statement, what is the return value of a call to that function?
+it returns nothing, with value of None
+
+# 9. How can you force a variable in a function to refer to the global variable?
+by not using it locally in an assignment statement
+
+# 10. What is the data type of None?
+NoneType
+
+# 11. What does the import areallyourpetsnamederic statement do?
+it would import a module called areallyourpetsnamederic, if it existed
+
+# 12. If you had a function named bacon() in a module named spam, how would you call it after importing spam?
+import spam
+
+spam.bacon()
+
+# 13. How can you prevent a program from crashing when it gets an error?
+with try and except statements, by putting the code that may have an error in a try clause
+
+# 14. What goes in the try clause? What goes in the except clause?
+the code that may have an error goes in the try clause
+what to do if that error happens goes in the except clause
+
+################################################################################################
+
+
+
+
+
+
 # Chapter 4
 
+# 1. What is []?
+an empty list
+
+# 2. How would you assign the value 'hello' as the third value in a list stored in a variable named spam? (Assume spam contains [2, 4, 6, 8, 10].)
```

Comment:
That's interesting. I read the problem differently, meaning you would replace the 3rd value, but I think you are right to insert it.